sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -30,6 +30,12 @@
 
 logger = logging.getLogger(__name__)
 
+if TYPE_CHECKING:
```

Comment:
Why do you want to use TYPE_CHECKING for these checks ?

Reply:
@andreyvelich I attempted to create a PR without adding this, but it failed the flake8 check in pre-commit check because the `HuggingfaceModelParams`, `HuggingFaceDatasetParams`, `S3DatasetParams`, and `HuggingFaceTrainerParams` classes are used in the definition of the `tune` function but were not defined or imported at that point. Therefore, I had to add the TYPE_CHECKING thing to pass the flake8 check. Do you have any alternative suggestions?

Here is the detailed message of the error:
```shell
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:178:45: F821 undefined name 'HuggingFaceModelParams'
sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:180:19: F821 undefined name 'HuggingFaceDatasetParams'
sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:180:47: F821 undefined name 'S3DatasetParams'
sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:182:38: F821 undefined name 'HuggingFaceTrainerParams'
```

Reply:
@andreyvelich Without adding these imports here, the flake8 check fails because the `tune` API references the following undefined names:
```shell
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:178:45: F821 undefined name 'HuggingFaceModelParams'
sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:180:19: F821 undefined name 'HuggingFaceDatasetParams'
sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:180:47: F821 undefined name 'S3DatasetParams'
sdk/python/v1beta1/kubeflow/katib/api/katib_client.py:182:38: F821 undefined name 'HuggingFaceTrainerParams'
```
To resolve this, I used TYPE_CHECKING to avoid actually importing these modules, while still passing the flake8 check, as these modules are only required when users choose to work with external models and datasets. Do you have any alternative suggestions?

Reply:
@helenxie-bit Can we follow the same as for Training Operator and remove fmt check for it: https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/training/api/training_client.py#L170-L185

Reply:
It looks better! I will modify according to this.